public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx

```diff
@@ -46,6 +58,16 @@ export function LabelFilterItem({ item, defaultOp, onChange, onDelete, onGetLabe
     return uniqBy([...selectedOptions, ...labelValues], 'value');
   };
 
+  /**
+   * !important here is necessary to show invalid border on all 4 sides of select.
+   * Without it, the invalid state is only visible on 3 sides as the right side is overridden in InputGroup.
+   */
+  const invalidClassNameOverride = invalidLabel
```

Comment:
Yes, totally! I was trying to look into it. I think we should fix it in `InputGroup`, because there we are removing left border from every child (starting with second child). Which would mean that we would have to check all children's props to see which one is invalid, and then keep the left border for those which are behind it. At least that was my idea. But I have created an issue and passed it to user essential team to look into. https://github.com/grafana/grafana/issues/56006